pkg/api/v1/types.go

```diff
@@ -1976,3 +1976,136 @@ type RangeAllocation struct {
+type AutoScaler struct {
+	TypeMeta   `json:",inline"`
+	ObjectMeta `json:"metadata,omitempty" description:"standard object metadata; see http://docs.k8s.io/api-conventions.md#metadata"`
+
+	// Spec defines the auto-scaler targets and thresholds.
+	Spec AutoScalerSpec `json:"spec,omitempty" description:"spec defines the auto-scaler targets and thresholds; http://docs.k8s.io/api-conventions.md#spec-and-status"`
+
+	// Status defines the actions the auto-scaler has taken.
+	Status AutoScalerStatus `json:"status,omitempty" description:"most recent status of an auto-scaler, populated by the auto-scale manager; http://docs.k8s.io/api-conventions.md#spec-and-status"`
+}
+
+// AutoScalerStatus provides the status of an auto-scaler.
+type AutoScalerStatus struct {
+	// Last autoscaler action trigger.
+	LastActionTrigger AutoScaleThreshold `json:"lastActionTrigger,omitempty" description:"threshold that triggered the last action"`
+
+	// Timestamp of the last autoscaler action.
+	LastActionTimestamp util.Time `json:"lastActionTimestamp,omitempty" description:"time of the last auto-scaler action"`
+}
+
+// AutoScalerSpec defines the auto-scaler targets and thresholds.
+type AutoScalerSpec struct {
+	// Thresholds holds a collection of AutoScaleThresholds that drive the auto-scaler.
+	Thresholds []AutoScaleThreshold `json:"thresholds,omitempty" description:"collection of thresholds that drive the auto-scaler"`
+
+	// MaxAutoScaleCount defines the max replicas that the auto-scaler can use.  This value must be
+	// >= MinAutoScaleCount.
+	MaxAutoScaleCount int `json:"maxAutoScaleCount,omitempty" description:"maximum replicas that the auto-scaler will scale upto"`
+
+	// MinAutoScaleCount defines the minimum number replicas that the auto-scaler can reduce to,
+	// 0 means that the application is allowed to idle.
+	MinAutoScaleCount int `json:"minAutoScaleCount,omitempty" description:"minumum replicas the auto-scaler will scale down to"`
+
+	// TargetSelector provides the resizeable target(s).  Right now this is a ReplicationController
+	// in the future it could be a job or any resource that implements resize.  If multiple targets
+	// are resolved by the selector the auto-scaler will resize the largest one.
+	TargetSelector map[string]string `json:"targetSelector,omitempty" description:"selector for the target the auto-scaler will resize"`
+
+	// Advisors defines a set of advisory sources that the auto-scaler
+	// uses to examine if thresholds are (b)reached.
+	Advisors []string `json:"advisors,omitempty" description:"list of auto-scale advisors"`
+}
+
+// AutoScaleThreshold is a behavior based on statistics used to drive the auto-scaler in scaling decisions.
+type AutoScaleThreshold struct {
+	// Type is the type of threshold being used, intention or value.
+	Type AutoScaleThresholdType `json:"type,omitempty" description:"auto-scale threshold type; intention or value based"`
```

Comment:
Good question - don't think it was anywhere but am not sure if I have the right answer to that. 

The types were as defined in the original autoscaler types PR  ttps://github.com/GoogleCloudPlatform/kubernetes/pull/5492, and when I started to use them, found a couple of missing configuration pieces to evaluate the thresholds - the value and duration in particular, which I added to the ```AutoScaleIntentionThresholdConfig``` struct. Given what the intention thresholds support now, the value based thresholds could simply be folded into the intention based ones (aka just need the intention based thresholds - value based becomes a subset).

Don't know the history/discussions leading to the initial autoscaler types PR, so maybe @pweil-  or @smarterclayton  can shed more light on that.  @pweil- / @smarterclayton ?